nntrainer/include/layer.h

```diff
@@ -479,20 +473,34 @@ class Layer {
   bool trainable;
 
   /**
-   * @brief     Gradient for the weights in this layer
-   * @note      The order of gradients should match the order in weights
+   * @brief     reserve memory for @a params and set @a param_size
+   * @exception std::invalid_argument when param_size is already set and
+   * shouldn't be changed again.
    */
-  std::vector<std::reference_wrapper<Tensor>> gradients;
+  void setParamSize(unsigned int psize) {
+
+    // @note Need opinion about this
```

Comment:
Lets enable for this now. Each layer functionality is fixed for now, later this can be changed.

Reply:
Ugh,, I found out that layer unittest need change. I'll just hold this for now and make a patch after this is merged